### PR TITLE
SonarLint cleanups

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

File: src/minoltamn_int.cpp [hunk 1]
```diff
@@ -1535,9 +1535,8 @@ static std::ostream& resolveLens0x34(std::ostream& os, const Value& value, const
     std::string model = getKeyString("Exif.Image.Model", metadata);
     std::string maxAperture = getKeyString("Exif.Photo.MaxApertureValue", metadata);
     long focalLength = getKeyLong("Exif.Photo.FocalLength", metadata);
-    std::string F2_8 = "760/256";
 
-    if (model == "SLT-A77V" && maxAperture == F2_8) {
+    if (model == "SLT-A77V" && maxAperture == "760/256") {
```
Comment (piponazo) [inline src/minoltamn_int.cpp:1539]: Hum, this change is a bit controversial. We lose some contextual information that was present in the variable name. If you really want to remove the variable I would add a comment explaining that it corresponds to F2.8

File: src/minoltamn_int.cpp [hunk 2]
```diff
@@ -1560,9 +1559,8 @@ static std::ostream& resolveLens0x80(std::ostream& os, const Value& value, const
     std::string model = getKeyString("Exif.Image.Model", metadata);
     std::string maxAperture = getKeyString("Exif.Photo.MaxApertureValue", metadata);
     long focalLength = getKeyLong("Exif.Photo.FocalLength", metadata);
-    std::string F4 = "1024/256";
 
-    if (model == "SLT-A77V" && maxAperture == F4 && inRange(focalLength, 18, 200)) {
+    if (model == "SLT-A77V" && maxAperture == "1024/256" && inRange(focalLength, 18, 200)) {
```
Comment (piponazo) [inline src/minoltamn_int.cpp:1563]: Same here about losing contextual information. People is used to the F numbers

**Review by piponazo — CHANGES_REQUESTED** [review]
Most of the changes look good to me, I am just hesitant about some of the changes and I left some comments with my observations.

File: src/minoltamn_int.cpp (hunk 1 above)
Comment (neheb) [reply src/minoltamn_int.cpp:1540]: Fixed.

File: src/minoltamn_int.cpp (hunk 2 above)
Comment (neheb) [reply src/minoltamn_int.cpp:1565]: Fixed.

File: src/bmffimage.cpp [hunk 3]
```diff
@@ -598,9 +598,8 @@ void BmffImage::parseTiff(uint32_t root_tag, uint64_t length, uint64_t start) {
     const size_t eof = std::numeric_limits<size_t>::max();  // impossible value for punt
     size_t punt = eof;
     for (size_t i = 0; i < exif.size() - 9 && punt == eof; ++i) {
-      if (exif.read_uint8(i) == exif.read_uint8(i + 1))
-        if (exif.read_uint8(i) == 'I' || exif.read_uint8(i) == 'M')
-          punt = i;
+      if (exif.read_uint8(i) == exif.read_uint8(i + 1) && (exif.read_uint8(i) == 'I' || exif.read_uint8(i) == 'M'))
```
[the remark this answers is not in the document]
Comment (neheb) [reply src/bmffimage.cpp]: Fixed.

**Review by piponazo — APPROVED** [review]

**Pull request merged**